pkg/identity/identitysync/okta.go

```diff
@@ -133,28 +136,36 @@ func (o *OktaSync) ListUsers(ctx context.Context) ([]identity.IDPUser, error) {
 
 	log.Debugw("listing all okta users")
 
-	users, res, err := o.client.User.ListUsers(ctx, &query.Params{})
-	if err != nil {
-		// try and log the response body
-		logResponseErr(log, res, err)
-		return nil, errors.Wrap(err, "listing okta users from okta API")
-	}
-
-	log.Debugw("listed all okta users")
-
-	for res.HasNextPage() {
-		var nextUsers []*okta.User
-		res, err = res.Next(ctx, &nextUsers)
+	oktaUsers := make(map[string]*okta.User)
+	for _, groupId := range strings.Split(o.groups.Get(), ",") {
```

Comment:
As commented, we should make this optional, so keep the old and the newer implementation. 

Funny, the page by page iterator has the same signature, in theory you could reuse that bit of code, but would suggest to duplicate for clarity

Reply:
BTW, makes sense to use a list as the gconfigField object seems to only support strings. Lets keep it simple. Also, we will likely only have one group here.

Reply:
Fixed.